WoWsShipBuilder.Common/Infrastructure/DataTransfer/AppNavigator.cs

```diff
@@ -0,0 +1,125 @@
+            AppPage.AccelerationCharts => AccelerationChartsBaseUrl + shipIndexes,
+            AppPage.ShipComparison => ShipComparisonBaseUrl + shipIndexes,
+            _ => throw new InvalidEnumArgumentException(),
+        };
+    }
+
+    private void GoToPage(AppPage destinationPage, List<ShipBuildContainer> containerList, string? shellIndex, AppPage? leavingPage, string? metricLabel)
+    {
+        int selectionCount = containerList.Count;
+        switch (selectionCount)
+        {
+            case > TransferLimit:
+                this.snackbar.Add($"Too many ships. Eligible ships: {selectionCount}. Max limit is {TransferLimit}.", Severity.Error);
+                return;
+            case 0:
+                this.snackbar.Add($"No ships available. Eligible ships: {selectionCount}.", Severity.Error);
+                return;
+            default:
+                this.LogMetrics(leavingPage, metricLabel);
+                this.sessionStateCache.SetBuildTransferContainers(containerList);
+                this.navManager.NavigateTo(GenerateDestinationUrl(destinationPage, containerList, shellIndex));
+                break;
+        }
+
+        // cant use await JsRuntime.InvokeAsync<object>("open", NavManager.BaseUri + url, "_blank"); to open in a new tab because builds are not carried over.
+    }
+
+    private void LogMetrics(AppPage? leavingPage, string? metricLabel)
+    {
+        if (string.IsNullOrEmpty(metricLabel))
+        {
+            return;
+        }
+
+        switch (leavingPage)
+        {
+            case AppPage.ShipStats:
+                this.metricsService.ShipStatsActions.WithLabels(metricLabel).Inc();
+                break;
+            case AppPage.BallisticCharts:
+                this.metricsService.BallisticChartsActions.WithLabels(metricLabel).Inc();
+                break;
+            case AppPage.AccelerationCharts:
+                this.metricsService.AccelerationChartsActions.WithLabels(metricLabel).Inc();
+                break;
+            case AppPage.ShipComparison:
+                this.metricsService.ShipComparisonActions.WithLabels(metricLabel).Inc();
+                break;
+            default:
+                return;
+        }
+    }
+
+    /// <summary>
+    /// Navigates to the specified destination page.
+    /// </summary>
+    /// <param name="destinationPage">The page to navigate to.</param>
+    /// <param name="containerList">The list of ship build containers to carry over.</param>
+    /// <param name="leavingPage">Optional. The page the user is leaving. Only needed for metrics.</param>
+    /// <param name="metricLabel">Optional.The label for the metric. Only needed for metrics.</param>
+    public void NavigateTo(AppPage destinationPage, IEnumerable<ShipBuildContainer> containerList, AppPage? leavingPage = null, string? metricLabel = null) => this.GoToPage(destinationPage, containerList.ToList(), null, leavingPage, metricLabel);
```

Comment:
Consider providing the values for the metric label as constant fields, maybe even in a static subclass of the AppNavigator so that they are accessible like `AppNavigator.Constants.ShipStatsToChartsLabel` or something like that